contributors/design-proposals/kep-template.md

```diff
@@ -0,0 +1,244 @@
+
+Software development provides a unique opportunity for an iterative approach to
+value delivery. Gathering user feedback is crucial for building high quality
+experiences and SIGs have the important responsibility of setting milestones
+for stability and feature completeness. Hopefully the content previously
+contained in [umbrella issues][] will be tracked in the `Graduation Criteria`
+section.
+
+[umbrella issues]: https://github.com/kubernetes/kubernetes/issues/42752
+
+### Section: Implementation History
+
+Major milestones in the life cycle of a KEP should be tracked in
+`Implementation History`. Major milestones might include
+
+- the `Summary` and `Motivation` sections being merged signaling SIG acceptance
+- the `Detailed Design` section being merged signaling agreement on a proposed
+  design
+- the date implementation started
+- the first Kubernetes release where an initial version of the KEP was available
+- the version of Kubneretes where the KEP graduated to general availability
+- when the KEP was retired or superseded
+
+### Section: Drawbacks
+
+Making trade offs is one of the fundamental tasks of software development. The
+`Drawbacks` section is to record what explicit choices have been made in the
+design for a KEP.
+
+### Section: Alternatives
+
+Similar to the `Drawbacks` section the `Alternatives` section is used to
+highlight and record other possible approaches to delivering the value proposed
+by a KEP.
+
+### Section: Unresolved Questions
+
+The `Unresolved Questions` section is used to parking lot issues not ready to be
+addressed before implementation begins.
+
+### Section: Mentors
+
+Mentors who can help a community member implement a KEP which follows its
+`Detailed Design` are crucial to scaling the Kubernetes project. Potential
+mentors can list their contact information using their preferred contact
+information in the `Mentors` section.
+
+## Graduation Criteria
+
+The first revision of the KEP template will be considered stable once
+
+- SIGs adopt the KEP process and template or fork the template
+- the KEP process has been used for at least two Kubernetes releases
+
+## Drawbacks
+
+- having many possible sections in a KEP may feel too much like the waterfall
+  approach to software development
+
+## Alternatives
+
+The KEP template is similar to the API proposal [design template][] but is
+slightly more general given that it targets the subset of information likely
+required for any significant change. Additionally
+
+- The `Overview subsequent versions` has been replaced with the
+  `Graduation Criteria` section of a KEP. It is intended that graduation
+  criteria are added incrementally as a KEP is implemented and stabilized
+- The `Motivation` and `Use Cases` sections have been collapsed within
+  the `Motivation` section of a KEP. It is expected that experience reports
+  will be the primary description of use cases
+- `Requirements` is replaced by `Detailed Design` within a KEP
```

Comment:
You mean that the Detailed Design section should contain the requirements, as well as the description of an implementation approach that meets those requirements?

We are going to have to expand the Detailed Design section, and/or find a good example document to refer to, but that can be done in subsequent iterations on the template.

Reply:
Following the latest iteration of the [Rust template][] split the `Detailed Design` into two sections: one explanation for users and one for contributors. 


[Rust template]: https://github.com/rust-lang/rfcs/blob/master/0000-template.md#reference-level-explanation